src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java

```diff
@@ -452,8 +452,10 @@ public void execute() throws GitException, InterruptedException {
 
                 StandardCredentials cred = credentials.get(url.toPrivateString());
                 if (cred == null) cred = defaultCredentials;
-                args.add(url);
-
+// JENKINS-56257 
+// prevent truncation of passward in the git url for clone				
+				args.add(url.toPrivateString());
```

Comment:
Please extend the CredentialsTest in this repository so that it can be used to test this case.  The CredentialsTest uses a json file that is local to your file system.  It reads plain text credentials from that json file and uses them to test authenticated git operations.

The CredentialsTest class is currently structured to require either a private key or a username / password.  You could extend that test to also accept an optional field like `credentialsEmbeddedInURL`.  When that field has the value `true`, then the URL should be passed through the test without applying a credential.

That testing technique allows the developer (you) and the maintainer (me) to evaluate credentials without sharing our credentials in the repository.